### DISCUSSION TITLE
Improved Add/Remove Device UX

### DISCUSSION THREAD

**Discussion by jarrettv** [zwave-js/zwave-js-ui#412] [Ideas]
I suggest we replace the drop-down controller actions menu with a simple **Add/Remove Device** button.

The user will choose from the 3 actions and click start or if they change their mind click close.

![image](https://user-images.githubusercontent.com/219295/106468847-12cf1300-6464-11eb-8d74-b19e51052492.png)

The user will see a countdown so they know how long they have to get to the device.

If there is an node included or excluded the user would see feedback on same dialog.

![image](https://user-images.githubusercontent.com/219295/106469187-9688ff80-6464-11eb-8d59-97222845146d.png)

The user can easily add another device by clicking start again.

We would need to relocate clean. I feel like that belongs in settings since it is a rare advanced action IMHO.

Thoughts?

**Comment by robertsLando** [comment]
I like that! Are you aable to submit a PR for that?

**Reply by AlCalzone** [reply]
I think it is important (also for the coming S2) to display how the node was included. It can happen that you press secure inclusion and something goes wrong, leaving the node included insecurely.

**Reply by AlCalzone** [reply]
@robertsLando I think we should move this into your repo

**Reply by jarrettv** [reply]
I can submit a PR after some further dev. Would you prefer the current drop down menu stays as is to get this going ASAP? Otherwise we need to relocate the other options somewhere.

**Reply by jarrettv** [reply]
![image](https://user-images.githubusercontent.com/219295/106472209-12387b80-6468-11eb-91ea-838277dea7e8.png)

**Reply by robertsLando** [reply]
@jarrettv I would remove the include/exclude actions from the dropdown and add a new button that opens that popup you showed here to handle the inclusion/exclusion process. The apis are there :)
